static/js/components/EducationForm.js

```diff
@@ -98,14 +102,59 @@ class EducationForm extends ProfileFormFields {
     </Grid>;
   };
 
-  handleSwitchClick(level){
+  entriesForLevel: Function = (level: string): number[] => {
+    const { profile } = this.props;
+    if ( _.isEmpty(profile.education) ) {
+      return [];
+    } else {
+      return Object.entries(profile.education).
+        filter(([, entry]: [string, EducationEntry]) => entry.degree_name === level).
+        map(([index,]) => Number(index));
+    }
+  };
+
+  closeDeleteAllEducationDialog: Function = (): void => {
     const {
-      ui: { educationDegreeInclusions },
+      setEducationDegreeLevel,
+      setShowEducationDeleteAllDialog,
+    } = this.props;
+    setEducationDegreeLevel('');
+    setShowEducationDeleteAllDialog(false);
+  };
+
+  deleteAllEducationEntriesForLevel: Function = (): void => {
+    const {
+      profile,
+      saveProfile,
+      ui,
+      ui: { educationDegreeLevel, educationDegreeInclusions },
       setEducationDegreeInclusions,
     } = this.props;
+    let toDelete = this.entriesForLevel(educationDegreeLevel);
+    let clone = _.cloneDeep(profile);
+    toDelete.forEach(index => clone.education.splice(index, 1));
```

Comment:
The `splice` in the loop here may have bad performance (though probably not noticeable) since all of the elements after the item being spliced are copied and this happens for each index being deleted. Why not do a `filter` here instead to remove the levels?